resolwe/flow/executors/__init__.py

```diff
@@ -112,63 +173,73 @@ def get_stdout(self):
         """Get process' standard output."""
         return self.stdout  # pylint: disable=no-member
 
-    def update_data_status(self, **kwargs):
-        """Update (PATCH) data object."""
-        data = Data.objects.get(pk=self.data_id)
-        for key, value in kwargs.items():
-            setattr(data, key, value)
+    def send_manager_command(self, cmd, expect_reply=True, **kwargs):
```

Comment:
I don't like that the public API exposes this low-level protocol allowing arbitrary commands with arbitrary keyword arguments as this makes it easier for typos to propagate unnoticed.

I would refactor this as follows:
* Rename this method to `_send_manager_command` (e.g. internal function).
* Introduce methods for sending different commands, which just forward arguments to `_send_manager_command`.
* If there is a lot to this protocol, you could also make a separate protocol handler class, which handles things like creating correct messages.

Reply:
This isn't really meant as any sort of public API, it's purely an internal helper. I renamed it into `_send_manager_command`.